src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
```

Comment:
Are you referring to ThreadLocal vs InheritableThreadLocal? (The only difference being that InheritableThreadLocal passes state from one thread to another thread if the second thread was created from the first thread)

It's a design decision whether or not passing state between threads should be explicit or implicit. The ThreadLocal in Java does it implicitly, so I guess that's the "standard" Java way. Another point of view is that we want to be able to trace requests without having to make lots of changes to userland code. Even more, to instruct developers to manually wrap all their threads and thread pools. In a multi-million-line code base, it's a pain to search-replace every place you need to wrap runnables and so on. (Unfortunately, that's what we have to do anyways, unless the Java standard library adopts something like ContinuationLocal and wraps ExecutorServices natively).

The underlying mechanism for passing state is the save() and restore() stuff, and the ExecutorService wrapper included in this library is simply a default implementation for thread pools. Likewise, InheritableThreadLocal could be thought of as a default implementation for passing state into new threads, I suppose?
Rxjava would have to add save()/restore() hooks to their library in order to achive the same thing.